### ISSUE TITLE
`gcs-object` creates 60 second time limit on unrelated changes

### ISSUE THREAD

**Issue by TV4Fun** [terraform-google-modules/terraform-google-secret#6]
This is probably a Terraform limitation, but the `gcs-object` module relies on a `google_storage_object_signed_url` data source, which Terraform creates at plan creation, and which has a default 60 second expiration. This means that if the plan is not applied within 60 seconds of creation, it will fail. This is a big problem if you're using something like Atlantis that requires you to plan and apply separately. This same process also seems to happen whether a plan change actually involves this resource or not. I suppose this is necessary because otherwise Terraform would have no way of knowing if there were any changes to the contents of the object or not, but it means that this constraint is now applied to every new plan from then on (not sure if `ignore_changes` would be useful here or not). There must be a better way to implement this. Is it at least possible to make Terraform not check for new object contents on every plan?

**Comment by morgante** [comment]
I don't know that is actually checks the object contents on plan? I think it's more that the URL will no longer be a valid data source after 60 seconds.

I think the only way to realistically fix this is to either increase the duration or refresh the plan before applying.

**Comment by TV4Fun** [comment]
Right, that's what I mean. It creates the URL on plan, but doesn't use it until apply. Ideally, it shouldn't create the URL until apply time. It might also be worthwhile to add a way to specify `ignore_changes` for this resource for content that you don't expect to change often and don't want being checked every time you plan and apply any change.

**Comment by morgante** [comment]
Unfortunately I'm not sure if ignore_changes would help in this case though, as the URL would still be expired and unusable.
